app/controllers/income_additional_controller.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class IncomeAdditionalController < StandardStepsController
+  def edit
+    @additional_income = current_snap_application.additional_income.map do |key|
+      "income_#{key}"
+    end
+
+    @step = step_class.new(
+      @additional_income.map { |k| [k, current_snap_application[k]] }.to_h,
+    )
+  end
+
+  private
+
+  def skip?
+    step_attrs.none? do |attr|
+      current_snap_application.
+        additional_income.
+        include?(attr.to_s.remove(/^income_/))
+    end
+  end
```

Comment:
What's your take on delegating to an additional, descriptive, well-named method to make this even easier to read?

Something like how I'm doing it over here: https://github.com/codeforamerica/michigan-benefits/pull/100/files#diff-7a0580538493197e22bd6c89261ac0a1R76

I had to do some digging on my branch over there to figure out *why* something necessitated skipping and it would have been helpful for me if I had known what the original developer in that controller was looking for.